op-conductor/conductor/service_test.go

```diff
@@ -129,7 +129,8 @@ func (s *OpConductorTestSuite) SetupTest() {
 	s.conductor.leaderUpdateCh = s.leaderUpdateCh
 
 	s.err = errors.New("error")
-	s.syncEnabled = false // default to no sync, turn it on by calling s.enableSynchronization()
+	s.syncEnabled = false   // default to no sync, turn it on by calling s.enableSynchronization()
+	s.wg = sync.WaitGroup{} // create new wg for every test in case last test didn't finish the action loop during shutdown.
```

Comment:
in the case that conductor shutdown happens before the last action loop gets executed (therefore wg.Done() won't be called).

We don't want to call wg.Wait() after shutdown because there's still chance shutdown happens before the action loop goroutine got scheduled. And this is purely a test issue, for production, it should exit as normal